server/controllers/logstreaming_controller.go

```diff
@@ -56,61 +59,57 @@ func newPullInfo(r *http.Request) (*PullInfo, error) {
 	if !ok {
 		return nil, fmt.Errorf("Internal error: no pull in route")
 	}
-
-	project, ok := mux.Vars(r)["project"]
-	if !ok {
-		return nil, fmt.Errorf("Internal error: no project in route")
-	}
 	pullNum, err := strconv.Atoi(pull)
 	if err != nil {
 		return nil, err
 	}
 
-	return &PullInfo{
-		Org:         org,
-		Repo:        repo,
-		Pull:        pullNum,
-		ProjectName: project,
+	return &pullInfo{
+		org:  org,
+		repo: repo,
+		pull: pullNum,
 	}, nil
 }
 
-func (j *LogStreamingController) GetLogStream(w http.ResponseWriter, r *http.Request) {
+// Gets the PR information from the HTTP request params
+func newProjectInfo(r *http.Request) (*projectInfo, error) {
 	pullInfo, err := newPullInfo(r)
 	if err != nil {
-		j.respond(w, logging.Error, http.StatusInternalServerError, err.Error())
-		return
+		return nil, err
 	}
-	//check db if it has pull project ret true or false(404)
-	exist, err := j.RetrievePrStatus(pullInfo)
-	if err != nil {
-		j.respond(w, logging.Error, http.StatusInternalServerError, err.Error())
-		return
+
+	project, ok := mux.Vars(r)["project"]
```

Comment:
is this a query param?

Reply:
yeah, it comes from the router `s.Router.HandleFunc("/jobs/{org}/{repo}/{pull}/{project}", s.JobsController.GetProjectJobs).Methods("GET").Name(ProjectJobsViewRouteName)`

Reply:
Why use query params for this? What about in the case that project isn't provided and project directory is instead? 

I still think we need to obfuscate this id.

Reply:
Yeah, this definitely doesn't handle the case with no projects. We do have a ticket to fix, so not sure if I should address it in here?